File: Day-1/Distinct-K/main.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;

string findKthDistinct(vector<string> &arr, int k)
{
    unordered_map<string, int> freq;

    for (auto str : arr)
    {
        freq[str]++;
    }

    for (auto str : arr)
    {
        if (freq[str] == 1)
        {
            if (k == 1)
                return str;
            k--;
        }
    }
    return "-1";
}

File: Day-1/Distinct-K/Main.java
import java.util.HashMap;
import java.util.Map;

public class Main {

    static String distinct_k(String[] arr, int k) {

        Map<String, Integer> freq = new HashMap<>();

        for (String str : arr) {
            freq.merge(str, 1, Integer::sum);
        }

        for (String str : arr) {
            if (freq.get(str) == 1) {
                if (k == 1)
                    return str;
                k--;
            }
        }
        return "-1";
    }
}